contracts/token/ERC20/ERC20.sol

```diff
@@ -241,25 +241,20 @@ contract ERC20 is Context, IERC20, IERC20Metadata {
     ) internal virtual {
         if (from == address(0)) {
             _totalSupply += amount;
-            unchecked {
-                // Overflow not possible: balance + amount is at most totalSupply + amount, which is checked above.
-                _balances[to] += amount;
-            }
-        } else if (to == address(0)) {
+        } else {
             uint256 fromBalance = _balances[from];
-            require(fromBalance >= amount, "ERC20: burn amount exceeds balance");
-            _totalSupply -= amount;
+            require(fromBalance >= amount, "ERC20: transfer amount exceeds balance");
             unchecked {
                 // Overflow not possible: amount <= fromBalance <= totalSupply.
                 _balances[from] = fromBalance - amount;
             }
+        }
+
+        if (to == address(0)) {
+            _totalSupply -= amount;
```

Comment:
this should be unchecked

Reply:
why?

Reply:
this cannot overflow:
- either the update is from 0 to 0, in which case we are removing what we just added → this is safe
- or we are updating from X to 0 (burn), in which case the amount burned is constrained by the from balance, which we know cannot be bigger than the totalSupply → this is also safe

Reply:
Yes, in fact this explanation is included above where it says:

```
// Overflow not possible: amount <= fromBalance <= totalSupply.
```

Reply:
Yes, in fact this explanation is included above where it says:

```
// Overflow not possible: amount <= fromBalance <= totalSupply.
```

Reply:
Still need to add unchecked to this line.